src/go/rpk/pkg/cli/acl/acl.go

```diff
@@ -72,17 +72,34 @@ group.
 
 PRINCIPALS
 
-All ACLs require a principal. A principal is composed of two parts: the type
-and the name. Within Redpanda, only one type is supported, "User". The reason
-for the prefix is that a potential future authorizer may add support for
-authorizing by Group or anything else.
+All ACLs require a principal or a role. A principal comprises a user and type.
+Within Redpanda, only the "User" type is supported. The "User" prefix ensures
+that potential future authorizers can add support for authorization by Group
+or some other type.
 
 When you create a user, you need to add ACLs for it before it can be used. You
 can create / delete / list ACLs for that user with either "User:bar" or "bar"
 in the --allow-principal and --deny-principal flags. This command will add the
 "User:" prefix for you if it is missing. The wildcard '*' matches any user.
 Creating an ACL with user '*' grants or denies the permission for all users.
 
+ROLES
+
+You can bind ACLs to a role. A role has only one part: the name. In contrast
+to principals, there is no need to supply the type. If a type-like prefix is
+present, it is considered part of the role name and processed accordingly.
+
+For example '--allow-role "foo"' specifies a role "foo", whereas
+'--allow-role "User:foo"' specifies a role "User:foo". In either case, the
```

Comment:
```suggestion
'--allow-role "User:foo"' specifies a role "foo", which is assigned to a specific user. In either case, the
```

Reply:
>which is assigned to a specific user

This is not correct. The point is that the prefix is _ignored_ here, but maybe the example just confuses things further?

Reply:
>ignored

More like "treated as text" rather than as type information